src/lib/parallel_scan/scan_state.md

```diff
@@ -0,0 +1,132 @@
+
+          _                _                 _                  _                 _                M7                  M6
+       _     _          _     _          M8     M8         M7     M7        M6      M6        M5        M5         M4      M4
+     B8  B8  _  _    B7  B7  B7  B7    B6  B6  B6  B6    B5  B5  B5  B5    B4  B4  B4  B4   B3   B3   B3   B3    B2   B2  B2   B2
+
+Block 9: Three transactions (B9's); complete three proofs (M6's) for the slots to be occupied in the second tree. Four proofs were added in the previous block and thereofre only three more needs to be done (maximum work given the constants is 7). For the thrid transaction, complete the next two proofs (B7's). The M6 proof from the last tree is returned as the ledger proof.
+
+          _                _                _                 _                  _                M9                M7
+       _     _        _      _           M9      _         M8     M8         M7     M7         M6      M6        M5        M5
+     B9  _  _  _    B8  B8  B9  B9    B7  B7  B7  B7    B6  B6  B6  B6    B5  B5  B5  B5    B4  B4  B4  B4   B3   B3   B3   B3
+
+Block 10: Four transactions, complete seven proofs (B7's and M7's)
+
+          _                  _                   _                _                 _                M10                M9
+       _     _          _         _          _      _         M9      M10       M8     M8         M7     M7         M6      M6
+     B10  _  _  _    B9  B10  B10  B10    B8  B8  B9  B9    B7  B7  B7  B7    B6  B6  B6  B6    B5  B5  B5  B5    B4  B4  B4  B4
+
+Block 11: three transactions, complete seven proofs (B8, B8, B9, B9, M8, M8, M9) in that order and return the M9 proof
+
+          _                _                      _                 _                  _                M11               M10
+       _     _         _         _          _         _         M11    M11         M9      M10       M8     M8         M7     M7
+     _  _  _  _    B10  B11  B11  B11    B9  B10  B10  B10    B8  B8  B9  B9    B7  B7  B7  B7    B6  B6  B6  B6    B5  B5  B5  B5
+
+ The number work that needs to be done is predetermined and depends on the number of slots occupied and which slot is occupied. The last slot of a tree requires only 1 proof to be done. This is because the total work that needs to be done is $2^{transaction\_capacity\_log\_2 - 1}$. The amount of work is always fixed based on the position of the slots being filled.
+ Having this constraint ensures:
+
+1. Multiple proofs are not emitted per update
+
+2. The merge node that is to be updated after adding proofs corresponding to its children is always empty. This allows us keep the number of trees to a minimum of $work\_delay+2$.
+
+The latency in this impl is as good as it can get because it emits a ledger proof for every $2^{transaction\_capacity\_log\_2}$ transactions.
```

Comment:
Oh I meant this in comparison with the previous one-big-tree implementation (for any throughput).  Just to elaborate on that, latency is the number of blocks the scan state takes to emit a new ledger proof. Earlier the tree had more leaves than $2^{transaction\_capacity\_log\_2}$ which is the maximum number of transactions per block. This increases the latency since there are more proofs to be done before emitting s ledger proof. I'll add this (and more) to the doc.